javascript/javascript3/week3/shoppingCartScript.js

```diff
@@ -0,0 +1,120 @@
+"use strict";
+console.log("shoppingCartScript is on");
+const userSection = document.createElement("section");
+const currencies = ["USD", "EUR", "DKK", "JPY"];
+class Product {
+  constructor(name, price, curr) {
+    this.name = name;
+    this.price = price;
+    this.currency = curr;
+  }
+  convertCurrency(curr) {
+    let conFactor; // conversion factor
+    switch (curr) {
+      case "USD":
+        conFactor = 0.15;
+        break;
+      case "EUR":
+        conFactor = 0.1343;
+        break;
+      case "DKK":
+        conFactor = 1.0;
+        break;
+      case "JPY":
+        conFactor = 17;
+        break;
+      default:
+        // default fallback for unlisted currency
+        return `Presented currency was not found in the database, default price is ${this.price} dkr`;
+    }
+    let displayPrice = this.price * conFactor; // THE MATH!!!!!!!!!!!!!!
+    return `${this.name} costs ${displayPrice} ${curr}`;
+  }
+}
+
+class ShoppingCart {
+  constructor() {
+    this.products = [];
+  }
+
+  addProduct(product) {
+    // Implement functionality here
+    this.products.push(product);
+  }
+
+  removeProduct(product) {
+    // Implement functionality here
+    this.products = this.products.filter((i) => i !== product);
```

Comment:
```suggestion
    this.products = this.products.filter((product) => product !== productToRemove);
```
But the risk of this is that it relies on object reference equality. 
If I create two flat-screen TVs and add those, then call to remove one, which one will be removed? The object that has the same memory reference. Perhaps it would be more correct to find the product by name but the exercise is vague on this.